### ISSUE TITLE
B905 not working since 0.0.284

### ISSUE THREAD

**Issue by ergoithz** [astral-sh/ruff#7669]
**Python 3.11.1**
**ruff 0.0.284...0.0.291**

Creating a file with the current tests https://github.com/astral-sh/ruff/blob/8165925e016a7e5f8fc038db834400288ca33c42/crates/ruff_linter/resources/test/fixtures/flake8_bugbear/B905.py#L3-L9  `ruff --select B905 ruff_b905.py` does not detect any rule violation.

```
$ ruff --version
ruff 0.0.291
$ cat ruff_b905.py 
# Errors
zip()
zip(range(3))
zip("a", "b")
zip("a", "b", *zip("c"))
zip(zip("a"), strict=False)
zip(zip("a", strict=True))
$ ruff --select B905 ruff_b905.py 
```

Last known working version was 0.0.283:
```
$ ruff --version
ruff 0.0.283
$ ruff --select B905 ruff_b905.py 
ruff_b905.py:2:1: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:3:1: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:4:1: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:5:1: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:5:16: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:6:5: B905 `zip()` without an explicit `strict=` parameter
ruff_b905.py:7:1: B905 `zip()` without an explicit `strict=` parameter
Found 7 errors.
```

**Comment by charliermarsh** [comment]
Thanks! B905 only applies when the target-version is >= Python 3.10 (since the `strict` parameter was added in Python 3.10). Running with `--target-version py310` does show the expected violations for me. How are you setting your target version?

**Comment by zanieb** [comment]
@ergoithz it seems likely you are affected by the change in the default target Python version https://github.com/astral-sh/ruff/blob/main/BREAKING_CHANGES.md#00283--0284

**Comment by ergoithz** [comment]
I'm genuinely impressed by how this tool can completely ignore python version ruff is installed with and instead pick an old hardcoded version as a target.

Regardless, since this design quirk is somewhat documented, I'll close the issue.

**Comment by charliermarsh** [comment]
For context: using the Python version that Ruff was installed with isn't sufficient to understand the version ranges supported by the project. You could be using Python 3.10 locally on a project that's designed to work with Python 3.7 up, in which case, using the current version would lead to breakages. In the future, we could fall back to the current version for projects that don't declare any such version, but it still wouldn't do what you describe if you had a hardcoded version in the project somewhere.

FWIW, the suggested way to do this is to set a `requires-python` in your `pyproject.toml` (this is also standards-compliant so other tools will respect it too), e.g.:

```toml
[project]
requires-python = ">=3.10"
```
